### PR TITLE
Clean up container docs

### PR DESCRIPTION
**Pull request by ru-fu**
Adding a bit of information on how containers work and updating the figures that show the container life cycle and that were still using the old hooks.
Plus a quick pass-through to replace other occurrences of the old hook names.

### CODE REVIEW

File: howto/container/launch.md [hunk 1]
```diff
@@ -1,21 +1,4 @@
...
-* Launch Android container
-* Execute the `prepare` hook provided by the installed addons if [launching a raw container](#raw-containers)
-* Execute the `backup` hook provided by the installed addons after the container is terminated
-
-![container-launch|690x401](upload://exw6GWcRvMzkIztcUIrizFg0oJz.png)
-
-The whole launch process will be successful only if all of the above steps succeed.
-
-If something goes wrong during the container launch process, the status of the container changes to `error` status. You can [view the available logs](https://discourse.ubuntu.com/t/view-the-container-logs/24329) from the container for diagnosing the root cause of the problem.
+Launching a container is normally done by another service over the REST API that the AMS service provides. However, the `amc` utility allows launching a container manually for experiments or development purposes. By default, the container will run headless and does not provide an entry point for a user to connect. Containers can be launched for either a registered application or image (see [Application containers vs. raw containers](https://discourse.ubuntu.com/t/managing-containers/17763#application-vs-raw)).
```
[the remark this answers is not in the document]
Comment (ru-fu) [reply howto/container/launch.md]: OK, that makes a lot more sense - I was wondering what all of this was supposed to mean. ;)
I rewrote it now, please check again.

File: explanation/containers.md [hunk 2]
```diff
@@ -1,12 +1,56 @@
...
+When we refer to containers in this documentation without specifying the container type, we mean regular containers.
+
+<a name="application-vs-raw"></a>
+## Application containers vs. raw containers
+
+Containers are based on either [applications](https://discourse.ubuntu.com/t/managing-applications/17760) or [images](https://discourse.ubuntu.com/t/provided-images/24185). That means that if you launch an application or an image, AMS automatically creates a container for it.
+
+Application containers, thus containers created when launching an application, run the full Android system. If the application is based on an Android app (an APK package), this app is launched after the system boots and monitored by the [watchdog](https://discourse.ubuntu.com/t/application-manifest/24197#watchdog). With the default configuration, you will only see the app and not the underlying Android system.
+
+Containers that are created when launching an image are called raw containers. They run only the base Android system.
```
[the remark this answers is not in the document]
Comment (aguadoenzo) [reply explanation/containers.md:20]: Oops my bad, messed up the wording in my original comment, you're right the bootstrap process only runs in the *base* container.

The point I was trying to get across is that I'm afraid a user might wonder about the difference between a `full Android system` and `only the base Android system`. Both raw and regular containers run Android in the same way. The APK is not really a part of the Android system itself.

In very simplified terms, application container = raw container + APK

File: explanation/containers.md [hunk 3]
```diff
@@ -1,12 +1,54 @@
...
+Base containers are temporary containers that are used when [bootstrapping an application](https://discourse.ubuntu.com/t/managing-applications/17760#bootstrap). They are automatically deleted when the application bootstrap is completed.
+
+When we refer to containers in this documentation without specifying the container type, we mean regular containers.
+
+<a name="application-vs-raw"></a>
+## Application containers vs. raw containers
+
+Containers are based on either [applications](https://discourse.ubuntu.com/t/managing-applications/17760) or [images](https://discourse.ubuntu.com/t/provided-images/24185). That means that if you launch an application or an image, AMS automatically creates a container for it.
+
+Application containers, thus containers created when launching an application, run the full Android system. If the application is based on an Android app (an APK package), this app is launched after the system boots and monitored by the [watchdog](https://discourse.ubuntu.com/t/application-manifest/24197#watchdog). With the default configuration, you will only see the app and not the underlying Android system.
```
Comment (aguadoenzo) [inline explanation/containers.md:18]: `you will only see the app and not the underlying Android system.`. Do you mean the Android launcher ?

As I understand it right now with `Android system`, it means the user won't have access to any aspect of Android, which isn't true, as he can interact with the Android system via adb and some other means

File: explanation/applications.md [hunk 4]
```diff
@@ -1,4 +1,4 @@
-Applications are one of the main objects AMS manages. A single application encapsulates one Android application and manages it within the cluster. It takes care of installing the supplied application package ([Android Package Kit - APK](https://en.wikipedia.org/wiki/Android_application_package)), to make it available to users. Further, AMS manages updates to existing applications, which includes allowing the operator to test new uploaded versions before making them available to any users.
+Applications are one of the main objects AMS manages. A single application encapsulates one Android app and manages it within the cluster. It takes care of installing the supplied app package ([Android Package Kit - APK](https://en.wikipedia.org/wiki/Android_application_package)), to make it available to users. Further, AMS manages updates to existing applications, which includes allowing the operator to test new uploaded versions before making them available to any users.
```
[the remark this answers is not in the document]
Comment (aguadoenzo) [reply explanation/applications.md]: That's a good idea, we definitely had some confusion in the past ourselves when using 'app' to define both the AMS application and an APK

File: explanation/applications.md [hunk 5]
```diff
@@ -30,20 +30,20 @@ A temporary base container will be created and used for the bootstrapping during
 
 In general, the bootstrap process goes through the following steps in order:
 
-![application-bootstrap|690x467](upload://haAJJ8p8ZEQXmsvrVb3HOHhl1io.png)
+1. Configure the network interface and gateway.
+2. Apply any pending Ubuntu system security updates.
+3. Install [addons](tbd) via the `pre-start` hook provided by each addon listed in the application manifest file.
```
[the remark this answers is not in the document]
Comment (aguadoenzo) [reply explanation/applications.md]: I think they should be split up if that's fine with you. When the `pre-start` hook is ran, the addon is already installed

File: explanation/containers.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (ru-fu) [reply explanation/containers.md:20]: Ah okay, now I understand your concern! You're right, this is misleading - updated. :)

File: explanation/containers.md (hunk 3 above)
Comment (ru-fu) [reply explanation/containers.md:18]: Not very familiar with Android terminology yet - but yes, I probably mean the launcher. ;) Namely that you can't just close the app and then see the "desktop".

**Review by aguadoenzo — APPROVED** [review]
LGTM !

**Pull request merged**